flanker/addresslib/address.py

```diff
@@ -390,27 +547,61 @@ def supports_routing(self):
 
     def full_spec(self):
```

Comment:
I implemented `full_spec` and `to_ace` separately because
1. I thought it would be less unexpected for anyone already using `full_spec` for it to return a valid address (especially since one is available) than for it to start throwing errors.
2. I think it will make updating downstream services easier.

There shouldn't be any ambiguity if we document it properly because `full_spec` always returns a bytestring just like it did before and `to_ace` and `to_unicode` always return Unicode.